src/GlobalSearch/GlobalSearchForm.js

```diff
@@ -1,43 +1,164 @@
-import React from 'react'
+import React, { Component, Fragment } from 'react'
 import '../index.scss'
 import '../lists.scss'
 import '../App.scss'
+import '../govStyles/govuk_frontend/all.scss'
 import PropTypes from 'prop-types'
 import ReactRouterPropTypes from 'react-router-prop-types'
+import { linkOnClick } from '../utils'
 
-const GlobalSearchForm = ({ searchText, handleSearchTextChange, handleSearch, history }) => (
-  <div>
-    <div className="pure-u-md-11-12 searchForm padding-top padding-bottom-large padding-left-30">
-      <label className="form-label" htmlFor="search-text">
-        Enter prisoner name or ID
-      </label>
-      <input
-        type="text"
-        className="width40 form-control"
-        id="search-text"
-        name="searchText"
-        value={searchText}
-        onChange={handleSearchTextChange}
-      />
-      <button
-        type="button"
-        id="search-again"
-        className="button margin-left margin-top-large"
-        onClick={() => {
-          handleSearch(history)
-        }}
-      >
-        Search again
-      </button>
-    </div>
-  </div>
-)
+class GlobalSearchForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      showFilters: false,
+    }
+  }
+
+  render() {
+    const {
+      handleSearchTextChange,
+      searchText,
+      handleSearch,
+      history,
+      genderFilter,
+      locationFilter,
+      handleSearchGenderFilterChange,
+      handleSearchLocationFilterChange,
+      clearFilters,
+    } = this.props
+
+    const toggleDetails = (event, clear) => {
+      event.preventDefault()
+      const { showFilters } = this.state
+      this.setState({
+        showFilters: !showFilters,
```

Comment:
According to [the react docs](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) you should use a different version of setState if you're referring to the current state.
`this.setState((state) => { showFilters: !state.showFilters }`